staging/src/k8s.io/endpointslice/reconciler.go

```diff
@@ -288,6 +293,22 @@ func (r *reconciler) reconcileByAddressType(logger klog.Logger, service *corev1.
 
 }
 
+func NewReconciler(client clientset.Interface, nodeLister corelisters.NodeLister, maxEndpointsPerSlice int32, endpointSliceTracker *endpointsliceutil.EndpointSliceTracker, topologyCache *topologycache.TopologyCache, eventRecorder record.EventRecorder) *Reconciler {
+	return &Reconciler{
+		client:               client,
+		nodeLister:           nodeLister,
+		maxEndpointsPerSlice: maxEndpointsPerSlice,
+		endpointSliceTracker: endpointSliceTracker,
+		metricsCache:         metrics.NewCache(maxEndpointsPerSlice),
+		topologyCache:        topologyCache,
+		eventRecorder:        eventRecorder,
+	}
+}
+
+func GetReconcilerName() string {
+	return controllerName
```

Comment:
shouldn't you make `controllerName`a field of the `Reconciler`object and pass it as a parameter of `NewReconciler`?

or how is people supposed to consume multiple reconciler and differentiate between them?

Reply:
lol, is in teh next commit